pkg/container/types/datalink.go

```diff
@@ -47,7 +47,7 @@ func ParseDatalink(fsPath string) (string, []int, string, error) {
 	// 2. get file extension
 	extension := filepath.Ext(u.Path)
 	switch extension {
-	case ".txt", ".csv":
```

Comment:
This is strange -- why datalink care anything of file extension?    It should not be part of datalink.

Reply:
try to replace this function with function.UrlToPath()